src/EditorFeatures/Core/Implementation/CodeDefinitionWindow/DefinitionContextTracker.cs

```diff
@@ -0,0 +1,240 @@
+        }
+
+        private async Task<ImmutableArray<CodeDefinitionWindowLocation>> GetContextFromPointAfterDelayAsync(
+            SnapshotPoint pointInRoslynSnapshot, CancellationToken cancellationToken)
+        {
+            // TODO: Does this allocate too many tasks - should we switch to a queue like the classifier uses?
+            await Task.Delay(TimeSpan.FromMilliseconds(250), cancellationToken).ConfigureAwait(false);
+
+            var document = pointInRoslynSnapshot.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+            {
+                return ImmutableArray<CodeDefinitionWindowLocation>.Empty;
+            }
+
+            return await GetContextFromPointAsync(document, pointInRoslynSnapshot.Position, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Internal for testing purposes.
+        /// </summary>
+        internal async Task<ImmutableArray<CodeDefinitionWindowLocation>> GetContextFromPointAsync(
+            Document document, int position, CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            if (!document.SupportsSemanticModel)
+            {
+                var goToDefinitionService = document.GetRequiredLanguageService<IGoToDefinitionService>();
+                var navigableItems = await goToDefinitionService.FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
+                if (navigableItems != null)
+                {
+                    var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
+
+                    var builder = new ArrayBuilder<CodeDefinitionWindowLocation>();
+                    foreach (var item in navigableItems)
+                    {
+                        if (navigationService.CanNavigateToPosition(workspace, item.Document.Id, item.SourceSpan.Start, cancellationToken))
+                        {
+                            var text = await item.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                            var linePositionSpan = text.Lines.GetLinePositionSpan(item.SourceSpan);
+
+                            if (item.Document.FilePath != null)
+                            {
+                                builder.Add(new CodeDefinitionWindowLocation(item.DisplayTaggedParts.JoinText(), item.Document.FilePath, linePositionSpan));
+                            }
+                        }
+                    }
+
+                    return builder.ToImmutable();
+                }
+
+                return ImmutableArray<CodeDefinitionWindowLocation>.Empty;
+            }
+            else
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = await SymbolFinder.FindSymbolAtPositionAsync(
```

Comment:
this feels like we're duplicating some work somewhere.  shoudl this move into whatever IGotoDef service we have, and have it expose the result that can then be consumed by other consumers as well as this.